src/txmempool.cpp

```diff
@@ -609,15 +610,14 @@ std::vector<uint256> CTxMemPool::mempoolDependenciesOf(const CTransactionBase& o
     return res;
 }
 
-CRawFeeRate CTxMemPool::avgFeeRateWithDeps(const uint256& rootTx, const bool certificatesAllowed, std::unordered_map<uint256, CRawFeeRate>& cache) const
+std::pair<CRawFeeRate, std::vector<uint256>> CTxMemPool::avgFeeRateWithDeps(const uint256& rootTx, const std::vector<CTxIn>& vin, const bool certificatesAllowed, std::unordered_map<uint256, CRawFeeRate>& cache) const
```

Comment:
Can be worth removing all the null-check on `entry` and leaving just one at function start? Something like:
```
if (!entry)
  return true;
```

Reply:
This comment should refer to function `trimToSize`.